### PR TITLE
Atomiks fix/options types

### PR DESCRIPTION
**Pull request by FezVrasta**
This is based on https://github.com/popperjs/popper-core/pull/1072

@trysound would you be able to figure out why my changes are not working as expected?

I'm getting the following:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ tests/flowtype/modifiers.js:19:52

Cannot call createPopper with object literal bound to options because:
 • string [1] is incompatible with string literal applyStyles [2] in property name of array element of property
   modifiers.
 • string [3] is incompatible with string literal offset [4] in property name of array element of property modifiers.

     tests/flowtype/modifiers.js
     16│
     17│ type ExtendedModifiers = StrictModifiers | CustomModifier;
     18│
     19│ createPopper<ExtendedModifiers>(reference, popper, {
     20│   modifiers: [
 [1] 21│     { name: 'offset', options: { offset: [0, ''] } }, // type error
 [3] 22│     { name: 'custom', options: { customOption: true } }, // no error
     23│   ],
     24│ });
     25│

     src/modifiers/applyStyles.js
 [2] 98│ }: Modifier<'applyStyles', {||}>);

     src/modifiers/offset.js
 [4] 65│ }: Modifier<'offset', Options>);



Found 2 errors
```

But the same logic works correctly on the Flow repl:
https://flow.org/try/#0PTAEAEDMBsHsHcBQiAuBPADgU1AeQNKgC8oA3qAHYCGAtlgFygDkAHkwDSgBuV0Arg1ABnFACcAlhQDmoAL6gAPmUq1BTNB268BjCnxoAjLKLkBuZAGNYFEaCqMCxZdTqNWmnvzUjRTOcmR0bFAAWQAeADlOADUAPidSJRdBKK0vRmjFWXMgnHxcJ3D3Th9JKXilIo1OPUNjWMtrWwNGfISVV2Y2Tk8dZh8-WWQgA

### CODE REVIEW

**Comment by atomiks** [comment]
You need to put `$Shape` around each modifier, not the whole:

```js
export type StrictModifiers =
  | $Shape<OffsetModifier>
  | $Shape<ApplyStylesModifier>
  | $Shape<ArrowModifier>
  | $Shape<HideModifier>
  | $Shape<ComputeStylesModifier>
  | $Shape<EventListenersModifier>
  | $Shape<FlipModifier>
  | $Shape<PreventOverflowModifier>
  | $Shape<PopperOffsetsModifier>;
```

File: src/types.js [hunk 1]
```diff
@@ -98,18 +108,30 @@ export type ModifierArguments<Options: Obj> = {
...
+  | OffsetModifier
+  | ApplyStylesModifier
+  | ArrowModifier
+  | HideModifier
+  | ComputeStylesModifier
+  | EventListenersModifier
+  | FlipModifier
+  | PreventOverflowModifier
+  | PopperOffsetsModifier
+>;
```
Comment (TrySound) [inline src/types.js:133]: Try to wrap each in Shape instead

**Comment by TrySound** [comment]
Oops, didn't see atomiks comment

**Comment by FezVrasta** [comment]
Thanks @atomiks and @TrySound , that did the trick!

**Pull request merged**